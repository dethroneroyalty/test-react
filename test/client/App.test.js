import chai, { expect } from "chai";
import chaiJestSnapshot from "chai-jest-snapshot";
import React from "react";
//import {BrowserRouter as Router, Link} from "react-router-dom";
import renderer from "react-test-renderer";
import ShallowRenderer from "react-test-renderer/shallow";

chai.use(chaiJestSnapshot);

function Link({ page, children }) {
	return <a href={page}>{children}</a>;
}

function MyComponent() {
	return (
		<div>
			<span className="heading">Title</span>
			<span className="heading">Title</span>
		</div>
	);
}

const shallowRenderer = new ShallowRenderer();
shallowRenderer.render(<MyComponent />);
const result = shallowRenderer.getRenderOutput();

describe("Bullshit", function() {
	it("renders", () => {
		const tree = renderer
			.create(<Link page="http://www.facebook.com"> Facebook </Link>)
			.toJSON();
		expect(tree).to.matchSnapshot();
	});
});
