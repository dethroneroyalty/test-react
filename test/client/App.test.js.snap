// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bullshit renders 1`] = `
<a
  href="http://www.facebook.com"
>
   Facebook 
</a>
`;
